Client/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -91,114 +162,79 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
+                make.left.greaterThanOrEqualTo(searchEngineScrollView).priority(1000)
             }
-            make.bottom.right.top.equalTo(self.searchEngineScrollView)
+            make.bottom.right.top.equalTo(searchEngineScrollView)
         }
     }
 
     private func layoutSearchEngineScrollView() {
         if !hasQuickSearchEngines { return }
         
-        let keyboardHeight = KeyboardHelper.defaultHelper.currentState?.intersectionHeightForView(self.view) ?? 0
+        let keyboardHeight = KeyboardHelper.defaultHelper.currentState?.intersectionHeightForView(view) ?? 0
         searchEngineScrollView.snp.remakeConstraints { make in
-            make.left.right.equalTo(self.view)
-            make.bottom.equalTo(self.view).offset(-keyboardHeight)
+            make.left.right.equalTo(view)
+            make.bottom.equalTo(view).offset(-keyboardHeight)
             make.height.equalTo(SearchViewControllerUX.engineButtonHeight)
         }
     }
     
-    var searchEngines: SearchEngines! {
-        didSet {
-            suggestClient?.cancelPendingRequest()
-
-            // Query and reload the table with new search suggestions.
-            querySuggestClient()
-
-            // Show the default search engine first.
-            if !tabType.isPrivate {
-                let ua = SearchViewController.userAgent ?? "FxSearch"
-                suggestClient = SearchSuggestClient(searchEngine: searchEngines.defaultEngine(), userAgent: ua)
-            }
-
-            // Reload the footer list of search engines.
-            reloadSearchEngines()
-            
-            layoutSuggestionsOptInPrompt()
-        }
-    }
-
-    private var quickSearchEngines: [OpenSearchEngine] {
-        var engines = searchEngines.quickSearchEngines
-
-        // If we're not showing search suggestions, the default search engine won't be visible
-        // at the top of the table. Show it with the others in the bottom search bar.
-        if tabType.isPrivate || !searchEngines.shouldShowSearchSuggestions {
-            engines?.insert(searchEngines.defaultEngine(), at: 0)
-        }
-
-        return engines!
-    }
-    
-    // If the user only has a single quick search engine, it is also their default one.
-    // In that case, we count it as if there are no quick suggestions to show
-    // Unless Default Search Engine is different than Quick Search Engine
-    private var hasQuickSearchEngines: Bool {
-        let isDefaultEngineQuickEngine = searchEngines.defaultEngine().engineID == quickSearchEngines.first?.engineID
-        
-        if quickSearchEngines.count == 1 {
-            return !isDefaultEngineQuickEngine
-        }
-
-        return quickSearchEngines.count > 1
-    }
-    
     private func layoutSuggestionsOptInPrompt() {
-        if tabType.isPrivate || !searchEngines.shouldShowSearchSuggestionsOptIn {
+        if tabType.isPrivate || !(searchEngines?.shouldShowSearchSuggestionsOptIn ?? false) {
```

Comment:
Instead of `?? false` use `searchEngines?.shouldShowSearchSuggestionsOptIn == false` so you don't have to unwrap.